### ISSUE TITLE
Docker is not starting

### ISSUE THREAD

**Issue by berserker5000** [docker/for-mac#6980]
### Description

MacOS: Ventura 13.5.2 (22G91) (intel chip)
Docker Desktop: unable to check because it is not loading even. (downloaded from docker.com on September 15, 2023)
Docker Engine: 24.0.6

After upgrading to the latest version of docker, the docker daemon cannot start and docker is not working anymore.

### Reproduce

docker ps

### Expected behavior

docker ps should show me all running containers, instead I'm getting 
`Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?`

### docker version

```bash
Client: Docker Engine - Community
 Version:           24.0.6
 API version:       1.43
 Go version:        go1.21.0
 Git commit:        ed223bc820
 Built:             Thu Aug 31 17:24:32 2023
 OS/Arch:           darwin/amd64
 Context:           default
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

### docker info

```bash
Client: Docker Engine - Community
 Version:    24.0.6
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.11.2-desktop.4
  compose: Docker Compose (Docker Inc.)
    Version:  v2.21.0-desktop.1
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.20
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.7
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
  scout: Command line tool for Docker Scout (Docker Inc.)
    Version:  0.24.1
```

### Diagnostics ID

unable to generate

**Comment by bsousaa** [comment]
Can you generate a diagnose from the terminal? https://docs.docker.com/desktop/troubleshoot/overview/#diagnose-from-the-terminal

**Comment by berserker5000** [comment]
@bsousaa thanks for the link. 
Diagnostics ID:     80E0E690-5DAF-42DF-9273-DDC7D6814A94/20230915151900

**Comment by faozimipa** [comment]
I have same issue 
Diagnostics ID:73380EE5-A7B4-4A59-B5C9-2984F517CB38/20230916090938

**Comment by nopparuts** [comment]
I have same issue
Diagnostics ID: F18A48FB-9895-48D0-8D62-A3A5F8E7168F/20230918024707

**Comment by cugarteblair** [comment]
I have the same issue
Diagnostics ID:     8D05BA32-0BB9-484E-AD17-D165843A6C23/20230918221039 (uploaded)

**Comment by Alex009** [comment]
same issue
E332082F-F924-4F5D-A78B-256C7B3D6F73/20230919040322

i also try to downgrade at 4.22.1 but nothing change :(

**Comment by berserker5000** [comment]
@Alex009 I tried to download previous versions of DD and had no luck with any of those. I have no idea what to do because I currently can't use Docker. Please let me know if you would find any version from the previous versions because I need docker for my work. Currently, I'm using AWS machines with docker to build images.

**Comment by Alex009** [comment]
@berserker5000 i am exactly in same situation. yesterday my old version of docker works, then i press update and now docker on my machine not work at all. 4.23.0 and 4.22.1 versions.
if you find some workaround - let me know too :)

**Comment by d3v4nt** [comment]
Same thing, diagnostic ID: `DC3984ED-3A23-45DE-AACB-60BF9199C0A5/20230919062023`

**Comment by thibauult** [comment]
Same issue: 

```
Diagnostics ID:     526926C4-7F24-4574-B82F-BB9958FC4649/20230919073237
```

**Comment by cugarteblair** [comment]
When I restored the version I had in my backups it comes up ok then quickly pops up a dialog that says: "Invalid virtual machine configuration. A directory sharing device configuration is invalid" I selected the Reset Docker to factory defaults and restarted. but it still does the same thing upon the restart.

**Comment by berserker5000** [comment]
@cugarteblair I tried your way and it didn't help. So I removed even the whole folder group.com.docker and after restarting DD is just blinked and crashed with no message.

**Comment by Alex009** [comment]
@berserker5000 i downgrade app to 4.22.1 and in `~/Library/Group Containers/group.com.docker/settings.json` i remove all `filesharingDirectories` - after it docker start successful.
thx @cugarteblair

**Comment by berserker5000** [comment]
Thanks all, I did downgrade to the 4.22.1 version, removed everything from `filesharingDirectories`, and intentionally put it as an empty string `""`  instead of empty list `[]` so docker asked me to reset everything to the default settings and it worked! Now I'm afraid to upgrade it to the newer version :) will wait for a couple of months before doing it.

**Comment by Hao-Wu** [comment]
delete ```filesharingDirectories ```  doesn't work for me.
macOS 13.5.2 and version 4.23.0

**Comment by RoNoLo** [comment]
Same problem. Update ran forever and now nothing works.

This is when I ran drocker inside the Docker.app

`/Applications/Docker.app/Contents/MacOS/install ; exit;
ron@MacBook ~ % /Applications/Docker.app/Contents/MacOS/install ; exit;

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.`

And this is what I found in hypervisor.error.json inside com.docker.docker

`{
  "type": "*errordef.dockerDesktopError",
  "component": "hypervisor",
  "originalError": {
    "type": "*errors.fundamental",
    "description": "Invalid virtual machine configuration. A directory sharing device configuration is invalid.",
    "stacktrace": [
      "github.com/docker/go-virtualization.(*VM).OnStartResult\n\tgithub.com/docker/go-virtualization@v0.0.0-20230829222754-3605d4b7b363/vm.go:439",
      "github.com/docker/go-virtualization.on_start_result\n\tgithub.com/docker/go-virtualization@v0.0.0-20230829222754-3605d4b7b363/callbacks.go:34",
      "_cgoexp_6a431636e773_on_start_result\n\t_cgo_gotypes.go:655",
      "runtime.cgocallbackg1\n\truntime/cgocall.go:315",
      "runtime.cgocallbackg\n\truntime/cgocall.go:234",
      "runtime.cgocallback\n\truntime/asm_arm64.s:1094",
      "runtime.systemstack_switch\n\truntime/asm_arm64.s:200",
      "runtime.cgocall\n\truntime/cgocall.go:167",
      "github.com/docker/go-virtualization._Cfunc_app_run\n\t_cgo_gotypes.go:310",
      "github.com/docker/go-virtualization.RunApplication\n\tgithub.com/docker/go-virtualization@v0.0.0-20230829222754-3605d4b7b363/app.go:18",
      "github.com/docker/pinata/mac/backend/cmd/com.docker.virtualization/vz.(*Configuration).RunVM\n\tgithub.com/docker/pinata/mac/backend/cmd/com.docker.virtualization/vz/run.go:144",
      "main.main.func1\n\tgithub.com/docker/pinata/mac/backend/cmd/com.docker.virtualization/main.go:96",
      "github.com/spf13/cobra.(*Command).execute\n\tgithub.com/spf13/cobra@v1.7.0/command.go:940",
      "github.com/spf13/cobra.(*Command).ExecuteC\n\tgithub.com/spf13/cobra@v1.7.0/command.go:1068",
      "github.com/spf13/cobra.(*Command).Execute\n\tgithub.com/spf13/cobra@v1.7.0/command.go:992",
      "main.main\n\tgithub.com/docker/pinata/mac/backend/cmd/com.docker.virtualization/main.go:118",
      "runtime.main\n\truntime/proc.go:250",
      "runtime.goexit\n\truntime/asm_arm64.s:1172"
    ]
  }
}`

Docker Desktop worked fine before the update.

**Comment by cugarteblair** [comment]
I should also add that every new attempt I restarted my OS. I was at factory defaults with completely reset data. I eventually went back to the current version which is the one I used that eventually worked. I am sure I "threaded a needle" I couldn't even see.
It would be nice if Docker developers could give some feedback

**Comment by intZero** [comment]
Thanks for your help all! I've literally just come back from holiday and Docker immediately died because of this issue! Have downgraded and the file reset has got me going again. Will warn everyone not to update.

**Comment by Dacascas** [comment]
I have the same issue. - `14296311-4CA8-4DD7-8B0A-21AACAF4FDBD/20230921190520`. Return to the 4.22, and I deleted all the `filesharingDirectories` in the `~/Library/Group Containers/group.com.docker/settings.json`

**Comment by cugarteblair** [comment]
> I have the same issue. - `14296311-4CA8-4DD7-8B0A-21AACAF4FDBD/20230921190520`. Return to the 4.22, and I deleted all the `filesharingDirectories` in the `~/Library/Group Containers/group.com.docker/settings.json`

Did that fix your issue?

**Comment by bhaidar** [comment]
I have the same issue.

Diagnose ID: A3A9B1CB-D775-483F-9A59-4D8F7B446A7B/20230922072143


Thanks

**Comment by o-l-o** [comment]
deleting all `filesharingDirectories` helped me, even without downgrading docker to previous version.

**Comment by o-l-o** [comment]
> deleting all `filesharingDirectories` helped me, even without downgrading docker to previous version.

In my case, the issue was that one of the sharing directories was on external (mountable) device. When the device wasn't mounted, the docker engine didn't start. When I mounted the device, the docker started without issues.

**Comment by bhaidar** [comment]
I had to remove and install Docker again. It seems working fine now.

**Comment by lorenrh** [comment]
Hi, 
We have a build that potentially fixes this issue, would you mind testing it?

- For Intel: https://desktop-stage.docker.com/mac/main/amd64/121765/Docker.dmg
- For M1 : https://desktop-stage.docker.com/mac/main/arm64/121765/Docker.dmg

Thanks!

**Comment by IvanSavoskin** [comment]
@lorenrh There was a similar problem, but with this image (for an Intel processor) everything works. Thanks!

**Comment by berserker5000** [comment]
@lorenrh Thanks for the new image, docker is working as expected but now k8s is not loading (waiting for 5 minutes atm). I will post an update in a couple of hours if it started or not.
Current status: docker-engine started and working, Kubernetes is starting

**Comment by Ivan-Strahovsky** [comment]
Had the same issue today, solution described here  helped me https://stackoverflow.com/questions/69552636/cannot-launch-docker-desktop-for-mac

**Comment by berserker5000** [comment]
After a couple of times restarting DD (intel, new image) everything worked for me. Docker is working, K8s is working.

**Comment by zhang-hongshen** [comment]
I met the same issue. Reinstalling docker didn't make sense.
MacBook Air M1, macOS Ventura 13.5.2.
`Diagnostics ID:     E1208A0D-718D-48BF-B7F4-35573610A51D/20230923083136`

**Comment by berserker5000** [comment]
Unfortunately, after adding the sharing directory your new images stopped working. Now even after modifying filesharingDirectories DD is not starting.

**Comment by PKC278** [comment]
> Hi, We have a build that potentially fixes this issue, would you mind testing it?
> 
> * For Intel: https://desktop-stage.docker.com/mac/main/amd64/121765/Docker.dmg
> * For M1 : https://desktop-stage.docker.com/mac/main/arm64/121765/Docker.dmg
> 
> Thanks!

m1 version docker, unable to enter the container command line through the Exec section.

**Comment by MihaelaStoica** [comment]
>  Hi, We have a build that potentially fixes this issue, would you mind testing it?
>  
>  * For Intel: https://desktop-stage.docker.com/mac/main/amd64/121765/Docker.dmg
>  * For M1 : https://desktop-stage.docker.com/mac/main/arm64/121765/Docker.dmg
>  
>  Thanks!

Please ensure that, before starting DD from this new image, there are no docker processes still running, in particular `com.docker.backend`.  @berserker5000 @zhang-hongshen @fdutenho please let us know if it still doesn't work after this.

**Comment by MihaelaStoica** [comment]
> m1 version docker, unable to enter the container command line through the Exec section.

We are aware of this issue on the build shared and we'll ensure it will be fixed before the next release.

**Comment by berserker5000** [comment]
> > Hi, We have a build that potentially fixes this issue, would you mind testing it?
> > 
> > * For Intel: https://desktop-stage.docker.com/mac/main/amd64/121765/Docker.dmg
> > * For M1 : https://desktop-stage.docker.com/mac/main/arm64/121765/Docker.dmg
> > 
> > Thanks!
> 
> Please ensure that, before starting DD from this new image, there are no docker processes still running, in particular `com.docker.backend`. @berserker5000 @zhang-hongshen @fdutenho please let us know if it still doesn't work after this.

I did double-check and force quit all docker processes, installed new image got next situation:
1) docker engine started as expected
2) adding shared folder, changing resource consuption in DD -> nothing changed, DD didn't restart after `apply and restart`
3) k8s didn't start and failed to stop when I tried to restart DD
4) force kill all docker processess
5) start DD -> everything started and applied all my changes.
So, basically I had to play around to make it running.

**Comment by sshindanai** [comment]
> > Hi, We have a build that potentially fixes this issue, would you mind testing it?
> > 
> > * For Intel: https://desktop-stage.docker.com/mac/main/amd64/121765/Docker.dmg
> > * For M1 : https://desktop-stage.docker.com/mac/main/arm64/121765/Docker.dmg
> > 
> > Thanks!
> 
> Please ensure that, before starting DD from this new image, there are no docker processes still running, in particular `com.docker.backend`. @berserker5000 @zhang-hongshen @fdutenho please let us know if it still doesn't work after this.

Working thanks!!

**Comment by jnsrnhld** [comment]
After deleting the version 4.23.0 from my mac, I deleted the whole `~/Library/Group/Containers/group.com.docker/` directory. Afterwards I installed the version prior to the broken update from [the release page](https://docs.docker.com/desktop/release-notes/#4221). This worked for me.

**Comment by Harrihauki** [comment]
I have the same issue but I'm using Ubuntu:

Description:	Ubuntu 22.04.3 LTS
Release:	22.04

Emptying filesharing gets the engine running, but Docker (perhaps obviously?) cannot see any files after that. Makes mounting sources etc impossible.

**Comment by lorenrh** [comment]
Hello,

This issue has been fixed in the latest 4.24.0 release. You can see the latest release notes [here](https://docs.docker.com/desktop/release-notes/).

I'll close the issue for now, but let us know if the issue persists.

**Comment by SpaceIgor** [comment]
@lorenrh маю цю проблему на 4.24
устанавлював попередні версії - нічого не допомогло - стопить роботу на проекті, як вирішити це питання?

**Comment by leonid-deriv** [comment]
4.23 did and does work (downgraded from 4.24). Kubernetes did not start on 4.24. Will update to MacOS to 14.0.

**Comment by SpaceIgor** [comment]
i cant built my project - on 4.24 and other ver

**Comment by leonid-deriv** [comment]
It did work after I recreated the cluster - it did not work when I have
upgraded from 4.23 to 4.24.

**Comment by Harrihauki** [comment]
> I have the same issue but I'm using Ubuntu:
> 
> Description: Ubuntu 22.04.3 LTS Release: 22.04
> 
> Emptying filesharing gets the engine running, but Docker (perhaps obviously?) cannot see any files after that. Makes mounting sources etc impossible.
> 
> Edit: Seems to be specifically Docker Desktop. My colleague says he does not have problems with only Docker Engine installed. Could be because filesharing is exclusively a setting in DD?

Still persists after updating.

**Comment by flavio1110** [comment]
4.24 didn't work for me, but the build mentioned [here](https://github.com/docker/for-mac/issues/6980#issuecomment-1731349344) worked just fine.

**Comment by jaschaio** [comment]
4.24.0 and above didn't work for me.

The last version that finally worked was 4.22.1

**Comment by berserker5000** [comment]
I also found a problem with the latest version: I can't add sharing paths to my docker engine.
Here is a video that doesn't work as expected.
https://streamable.com/0ffsbi

**Comment by vishalzambre** [comment]
Diagnostics ID: 7357B5CE-FFC5-48D8-AF74-6F0EFE562645/20231116073805

**Comment by chengyjonathan** [comment]
I'm running into the same issue

macOS Moneterey
Version 12.3
Apple M1 Max Chip

Client:
 Version:    24.0.6
 Context:    desktop-linux
 Debug Mode: false

Running into this same issue
Diagnostics ID:  60088DDC-B482-498B-A586-B6A272093405/20231116181516

**Comment by jaschaio** [comment]
Just updated to `4.25.1` on macOS Monterey 12.1 with Apple M1 Max and it still failing with the same error

**Comment by remixer-dec** [comment]
Also running into this on 4.25.1, updating docker was a mistake

**Comment by bsousaa** [comment]
@remixer-dec @jaschaio can you try rebooting your machine and start Docker Desktop again?

**Comment by bblacey** [comment]
@bsousaa FYI, I shutdown every night and reboot each morning but Docker Desktop still does not launch at startup so I don't think that will fix @remixer-dec or @jaschaio's issue either.  I am currently running version v4.25.1.  Version 4.25.0 didn't launch at startup either.

**Comment by jaschaio** [comment]
I already tried that plus resetting to factory defaults but no difference.

**Comment by trensen** [comment]
I removed all entries in `filesharingDirectories`, rebooted and it's not launching on startup still.

MacOS Sonoma 14.1.1
Apple M2 Pro Chip

Client: "Docker Desktop 4.25.2 (129061) is currently the newest version available."

Diagnostic `75C4DEF8-BABB-47C8-B733-484A66183273/20231124054020`

**Comment by bsousaa** [comment]
@trensen can you factory reset?

**Comment by trensen** [comment]
> @trensen can you factory reset?

I just did and rebooted, but it's still not automatically starting. Also tried a "Clean / Purge data", but this didn't do the trick either unfortunately...

**Comment by bsousaa** [comment]
@trensen, are you talking about automatic start after login? See this issue https://github.com/docker/for-mac/issues/7052

**Comment by trensen** [comment]
> @trensen, are you talking about automatic start after login? See this issue #7052

Thanks, hopefully it's fixed in the next version 🤞

**Comment by shiruken1** [comment]
Just upgraded to 4.25.2 and facing this issue. M1 chip OSX 12.2.1

**Comment by trensen** [comment]
I can confirm that updating to 4.26.0 fixes the issue for me

**Comment by Luckyfella73** [comment]
I wasn't able to start Docker Desktop either after pressing the restart-and-update button. Had Docker installed as Brew cask. Uninstalling via Brew `brew uninstall --cask docker` and installing a fresh copy of Docker `brew install --cask docker` worked for me. Had to manually delete `/usr/local/bin/docker-index` file though (got an error when trying to install Docker in the first run).

M1
macOS 12.6.3

**Comment by eugeneYWang** [comment]
> Just upgraded to 4.25.2 and facing this issue. M1 chip OSX 12.2.1

4.24 had no issue at all. M1 MAX macOS

**Comment by Emanuele-Spatola** [comment]
Unable to run docker after the last update. Please help! 🙏
Mac OS: 12.7.3
Docker Desktop: 4.27.2
Diagnostics ID:     7B3C9900-32C3-446B-858F-5858B84DA853/20240307085440

**Comment by vladmiller** [comment]
```[2024-04-10T15:37:44.999012000Z][com.docker.virtualization][I] VM has started
[2024-04-10T15:37:45.062574000Z][com.docker.virtualization][I] VM has stopped: Invalid virtual machine configuration. The storage device attachment is invalid.
[2024-04-10T15:37:45.062643000Z][com.docker.virtualization][I] Searching for crashes after error Invalid virtual machine configuration. The storage device attachment is invalid.
[2024-04-10T15:37:55.075127000Z][com.docker.virtualization][I] pairConnections returned context canceled
[2024-04-10T15:37:55.075041000Z][com.docker.virtualization.ipc][I] shutting down e3c1a178-VirtualizationAPI for component com.docker.virtualization on <HOME>/Library/Containers/com.docker.docker/Data/virtualization.sock on context done
[2024-04-10T15:37:55.075282000Z][com.docker.virtualization][I] sendConnections returned <nil>
[2024-04-10T15:37:55.076012000Z][com.docker.virtualization][I] Requesting VM termination
[2024-04-10T15:37:55.076089000Z][com.docker.virtualization.ipc][W] shutting down e3c1a178-VirtualizationAPI for component com.docker.virtualization: close unix <HOME>/Library/Containers/com.docker.docker/Data/virtualization.sock: use of closed network connection
[2024-04-10T15:37:55.076049000Z][com.docker.virtualization][I] runAPI returned <nil>
[2024-04-10T15:37:55.086585000Z][com.docker.virtualization][I] vz.RunApplication returned
[2024-04-10T15:37:55.087026000Z][com.docker.virtualization][E] com.docker.virtualization: Invalid virtual machine configuration. The storage device attachment is invalid.
[goroutine 1 [running, locked to thread]:
[main.main()
[       mac/backend/cmd/com.docker.virtualization/main.go:143 +0xa90
```

The thing crashes after a while when using Earthly to build some images. Docker just hangs, I can't remove any containers, can refresh or whatever. When I try to restart docker it would just crash and offer to reset to factory defaults.

Version: 4.28.0 (139021)
Engine: 25.0.3
